### PR TITLE
Use node display name in messages 

### PR DESCRIPTION
**Pull request by vmunishwar**

### CODE REVIEW

File: pkg/yamltemplate/filetests/if-err-too-many-end.yml [hunk 1]
```diff
@@ -12,4 +12,4 @@ metadata:
 ERR: 
 - unexpected end
     stdin:8 |     #@ end
-    stdin:? | __ytt_tplXXX_start_node(11) # *yamlmeta.Document (generated)
\ No newline at end of file
+    stdin:? | __ytt_tplXXX_start_node(11) # document (generated)
```
Comment (pivotaljohn) [inline pkg/yamltemplate/filetests/if-err-too-many-end.yml]: In this case, the reader is _at least_ an advanced ytt user, if not a maintainer. Also, internal details abound in this output. By my eye, it's appropriate to present the Go type, here.

Comment (vmunishwar) [reply pkg/yamltemplate/filetests/if-err-too-many-end.yml]: Thanks for the review. I have made the required changes.

File: pkg/schema/schema.go [hunk 2]
```diff
@@ -252,9 +252,12 @@ func inferTypeFromValue(value interface{}, position *filepos.Position) (yamlmeta
 		return &ScalarType{ValueType: *new(bool), defaultValue: typedContent, Position: position}, nil
 	case nil:
 		return nil, nil
+	case yamlmeta.Node:
+		return nil, fmt.Errorf("Expected value '%s' to be a map, array, or scalar, but was %s", value, typedContent.DisplayName())
+	default:
+		return nil, fmt.Errorf("Expected value '%s' to be a map, array, or scalar, but was %T", value, value)
 	}
```
Comment (cari-lynn) [inline pkg/schema/schema.go]:
```suggestion
```
Style comment: newline can be removed

File: pkg/yamltemplate/starlark_fragment.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ const (
 	starlarkFragmentType = "yamlfragment"
 )
 
+// StarlarkFragment is an adapter for yamlmeta.Node values in Starlark code
```
Comment (cari-lynn) [inline pkg/yamltemplate/starlark_fragment.go:20]: Great to see the godocs comment added here! It will definitely help with understanding this code.

**Review by cari-lynn — APPROVED** [review]
This is definitely an improvement to error messages. And it is nice to see that there are many places we will be able to use the display name.

File: pkg/schema/schema.go (hunk 2 above)
Comment (vmunishwar) [reply pkg/schema/schema.go]: Thanks for the review. Committed the changes.

**Review by gcheadle-vmware — APPROVED** [review]
LGTM! Great work

**Pull request merged**